### DISCUSSION TITLE
Does the dragonfly support the  redis cluster mode ?

### DISCUSSION THREAD

**Discussion by tiandong19860806** [dragonflydb/dragonfly#445] [Q&A] [answered]
Hi  all,

       Does the dragonfly support the  redis cluster mode ?

such like below:

(1) dragonfly cluster is three nodes
spring.redis.cluster.nodes=
192.168.4.21:7001, 192.168.4.21:7002,
192.168.4.22:7001, 192.168.4.22:7002,
192.168.4.23:7001, 192.168.4.23:7002

(2) java code to connect dragonfly cluster (like protecol : redist cluster)
if(!StringUtil.isEmpty(password)){
	jedisCluster = new JedisCluster(hostAndPorts, connectionTimeout, soTimeout, maxAttempts, password, poolConfig);
}
else {
	jedisCluster = new JedisCluster(hostAndPorts, connectionTimeout, soTimeout, maxAttempts, poolConfig);
}

**Comment by tiandong19860806** [comment]
As test below,

(1) redis-cli connect dragonfly server is success
![image](https://user-images.githubusercontent.com/12596666/198978217-fee030b6-76a1-442b-95a5-10c24baa5009.png)

(2) but when I want use cluster mode to connect dragonfly is failed. 
**Note: please confirm the dragonfly whether has solution to deploy cluster mode as the redis cluster ?**
 
![image](https://user-images.githubusercontent.com/12596666/198978648-84173085-260a-418d-8d98-776be5260aa7.png)
![image](https://user-images.githubusercontent.com/12596666/198978893-dbce40f8-8c11-49df-aa0b-f5f273ea6614.png)

**Comment by romange** [comment, marked as answer]
Dragonfly currently does not support cluster mode commands.
We plan on implementing the emulation for a single node setup. See https://github.com/dragonflydb/dragonfly/issues/332

**Reply by tiandong19860806** [reply]
good work, thanks.

**Reply by tiandong19860806** [reply]
Hi @romange ,

I follow your suggest as link https://github.com/dragonflydb/dragonfly/issues/332,

I use the command flag 'cluster-enabled' and value 'emulated', 

But when I run dragonfly docker that it happen error as below:

command as below : 
docker run --cluster-enabled emulated  --network=host --ulimit memlock=-1 --rm dragonfly --port 16379 --bind 0.0.0.0 --requirepass password#123  --dir  /$APP_DATA_HOME  --keys_output_limit 100 

![图片](https://user-images.githubusercontent.com/12596666/199154361-9be22ceb-f03d-4b29-97d6-4d296230c50c.png)

**Comment by odedponcz** [comment]
In other words since Dragonfly can scale vertically, this will allow you to pack a few cluster nodes into a single Dragonfly instance.
You will not need to change your client code to support this.

**Comment by romange** [comment]
Sorry i was not clear. This feature is not implemented yet, it's an open
task.
